docs/advanced-topics/advanced-losses-optimizers-and-poolers.md

```diff
@@ -78,6 +78,13 @@ Each color represents a different class. You can see how all of the loss functio
 but struggle to separate the green, blue, pink, purple and red classes,
 with `TripletMarginLoss` sperarating them the least, and `ArcFaceLoss` separating them the most.
 
+### CosineSimilarityLoss
+
+`CosineSimilarityLoss` is a contrastive loss function,
+the loss is calculated by comparing the cosine similarity of two embeddings against their ground truth cosine similarity.
+The goal of `CosineSimilarityLoss` is to minimize the MSE (mean squared error) between document pairs and their ground truth
+in order to improve the similarity between two semantically related sentences.
```

Comment:
```suggestion
in order to improve the representation of similarity between two semantically related documents (e.g., sentences or images).
```
Maybe something like this, since it does not only work for text.